core/src/scene/drawRule.cpp

```diff
@@ -228,6 +234,41 @@ bool DrawRule::operator<(const DrawRule& _rhs) const {
     return style < _rhs.style;
 }
 
+bool StyleParam::parseVec2(const std::string& _value, const std::vector<std::string>& _allowedUnit, std::pair<float, float>& _vec2) {
+    if (_value.empty()) {
+        return false;
+    }
+
+    std::string value = _value;
+
+    // replace all unit occurences
+    for (auto& unit : _allowedUnit) {
+        auto i = value.find(unit);
+        // TODO: conversion
+        while (i != std::string::npos) {
```

Comment:
This step is only replacement to have a clean string though, I added an early exit if this step fails.